datafusion/physical-plan/src/sorts/sort.rs

```diff
@@ -1205,9 +1221,10 @@ impl ExecutionPlan for SortExec {
         self: Arc<Self>,
         children: Vec<Arc<dyn ExecutionPlan>>,
     ) -> Result<Arc<dyn ExecutionPlan>> {
-        let new_sort = SortExec::new(self.expr.clone(), Arc::clone(&children[0]))
+        let mut new_sort = SortExec::new(self.expr.clone(), Arc::clone(&children[0]))
             .with_fetch(self.fetch)
             .with_preserve_partitioning(self.preserve_partitioning);
+        new_sort.filter = Arc::clone(&self.filter);
```

Comment:
I missed this for a while and spent an hour trying to figure out why my test was failing. IMO we should have a test that enforces the invariant that `ExecutionPlan::with_new_children(Arc::clone(&node), node.children()) == node`